src/Symfony/Bundle/FrameworkBundle/Resources/config/routing.xml

```diff
@@ -54,7 +54,7 @@
         <service id="router.default" class="Symfony\Bundle\FrameworkBundle\Routing\Router">
             <tag name="monolog.logger" channel="router" />
             <tag name="container.service_subscriber" id="routing.loader" />
-            <argument type="service" id="Psr\Container\ContainerInterface" />
+            <argument type="service" id="service_container" />
```

Comment:
is this correct? It seems previously a smaller service locator was injected since that `Router` is a `ServiceSubscriberInterface`.

@nicolas-grekas this is very related to our conversation from this morning? :smile:

Reply:
I confirm this one gets the main container, that's why controllers must be public.

Reply:
So I just check this on my 4.4 app:

before:

```
Information for Service "router.default"
========================================

 This Router creates the Loader only when the cache is empty.

 ---------------- -------------------------------------------------------------- 
  Option           Value                                                         
 ---------------- -------------------------------------------------------------- 
  Service ID       router.default                                                
  Class            Symfony\Bundle\FrameworkBundle\Routing\Router                 
  Tags             monolog.logger (channel: router)                              
                   container.service_subscriber (id: routing.loader)             
  Calls            setConfigCacheFactory                                         
  Public           no                                                            
  Synthetic        no                                                            
  Lazy             no                                                            
  Shared           yes                                                           
  Abstract         no                                                            
  Autowired        no                                                            
  Autoconfigured   no                                                            
  Arguments        Service(.service_locator.KThzMvS.router.default)              
                   /var/www/x/symfony/app/config/routing_dev.yml  
                   Array (7 element(s))                                          
                   Service(router.request_context)                               
                   Service(parameter_bag)                                        
                   Service(monolog.logger.router)                                
                   en                                                            
 ---------------- -------------------------------------------------------------- 
```

and now with this change:

```
Information for Service "router.default"
========================================

 This Router creates the Loader only when the cache is empty.

 ---------------- -------------------------------------------------------------- 
  Option           Value                                                         
 ---------------- -------------------------------------------------------------- 
  Service ID       router.default                                                
  Class            Symfony\Bundle\FrameworkBundle\Routing\Router                 
  Tags             monolog.logger (channel: router)                              
                   container.service_subscriber (id: routing.loader)             
  Calls            setConfigCacheFactory                                         
  Public           no                                                            
  Synthetic        no                                                            
  Lazy             no                                                            
  Shared           yes                                                           
  Abstract         no                                                            
  Autowired        no                                                            
  Autoconfigured   no                                                            
  Arguments        Service(service_container)                                    
                   /var/www/x/symfony/app/config/routing_dev.yml  
                   Array (7 element(s))                                          
                   Service(router.request_context)                               
                   Service(parameter_bag)                                        
                   Service(monolog.logger.router)                                
                   en                                                            
 ---------------- -------------------------------------------------------------- 
```

So its `Service(.service_locator.KThzMvS.router.default)` vs `Service(service_container)`

Reply:
Oh, my bad. Tag `container.service_subscriber`.
Should be reverted indeed.
@fancyweb can you please check the other changes looking for the same tag?

Reply:
I reverted everything. We actually do not rely on those aliases internally. Thanks for catching my mistake @dmaicher.